File: python/sort_wstaw.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def sort_wstaw(lista):
    """wersja liniowa"""
    for i in range(1, len(lista)):
        el = lista[i]
        k = i - 1
        while k >= 0 and lista[k] < el:  # wyszukanie pozycji
            lista[k + 1] = lista[k]  # przesuwanie elementów w góre tabeli
            k -= 1
        lista[k + 1] = el  # wstawianie elementów
    return lista


def wyszukaj_bin(lewy, prawy, lista, el):

    lewy, prawy = 0, len(lista) - 1
    while lewy < prawy:
        sr = floor((lewy + prawy) / 2)
        if el <= lista[sr]:
            prawy = sr
        else:
            lewy = sr + 1
    if lista[lewy] == el:
        return lewy

    return -1


def sort_wstaw_bin(lista):
    for i in range(1, len(lista)):
        el = lista[i]
        k = wyszukaj_bin(lewy, prawy, lista, el)
        lista =


def main(args):
    lista = [4, 3, 7, 0, 2, 3, 1, 9]
    print(lista)
    print(sort_wstaw(lista))
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/sort_wstaw.cpp
#include <iostream>

using namespace std;

void sort_wstaw(int t[], int n)
{
    int el=0;
    int k=0;
    
    for(int i=1;i<n;i++)
    {
        el = t[i];
        k = i-1;
        
        while(k>=0 && t[k] > el)
        {
            t[k+1] = t[k];
            k--;
        }
        t[k+1] = el;
    }
}
void drukuj(int tab[], int n)
{
    for(int i=0; i<n; i++)
    {
        cout<<tab[i]<<" ";
    }
}

int main(int argc, char **argv)
{
    int ile = 8;
    int lista[] = {4, 3, 7, 0, 2, 3, 1, 9};

    drukuj(lista, ile);

    cout<<endl;

    sort_wstaw(lista, ile);
    drukuj(lista, ile);
    
    return 0;
}
